src/rust/ide/view/graph-editor/src/lib.rs

```diff
@@ -976,11 +976,19 @@ impl GraphEditorModelWithNetwork {
 
             // === Visualizations ===
 
+            let vis_changed    =  node.model.visualization.frp.visualisation.clone_ref();
             let vis_visible    =  node.model.visualization.frp.set_visibility.clone_ref();
             let vis_fullscreen =  node.model.visualization.frp.enable_fullscreen.clone_ref();
+
             vis_enabled        <- vis_visible.gate(&vis_visible);
             vis_disabled       <- vis_visible.gate_not(&vis_visible);
 
+            // Ensure the graph editor knows about internal changes to the visualisation. If the
+            // visualisation changes that should indicate that the old one has been disabled and a
+            // new one has been enabled.
+            output.visualization_disabled          <+ vis_changed.constant(node_id);
+            output.visualization_enabled           <+ vis_changed.constant(node_id);
+            
```

Comment:
I'd rather opt for making "visualization_changed" a part of the GraphEditor API, and do tricks on the integration level. But this can take a bit of time, so for now please make an issue for refactoring this, and put here TODO with issue number.

Reply:
Added the task #896  and the todo in the doc.